fury/actors/peak.py

```diff
@@ -0,0 +1,372 @@
+from fury.colormap import boys2rgb, colormap_lookup_table, orient2rgb
+from fury.shaders import attribute_to_actor, load, shader_to_actor
+from fury.utils import apply_affine, numpy_to_vtk_colors, numpy_to_vtk_points
+from vtk.util import numpy_support
+
+
+import numpy as np
+import vtk
+
+
+class PeakActor(vtk.vtkActor):
+    """
+    VTK actor for visualizing slices of ODF field.
+
+    Parameters
+    ----------
+    directions : ndarray
+        Peak directions. The shape of the array should be (X, Y, Z, D, 3).
+    indices : tuple
+        Indices given in tuple(x_indices, y_indices, z_indices)
+        format for mapping 2D ODF array to 3D voxel grid.
+    values : ndarray, optional
+        Peak values. The shape of the array should be (X, Y, Z, D).
+    affine : array, optional
+        4x4 transformation array from native coordinates to world coordinates.
+    colors : None or string ('rgb_standard') or tuple (3D or 4D) or
+             array/ndarray (N, 3 or 4) or array/ndarray (K, 3 or 4) or
+             array/ndarray(N, ) or array/ndarray (K, )
+        If None a standard orientation colormap is used for every line.
+        If one tuple of color is used. Then all streamlines will have the same
+        color.
+        If an array (N, 3 or 4) is given, where N is equal to the number of
+        points. Then every point is colored with a different RGB(A) color.
+        If an array (K, 3 or 4) is given, where K is equal to the number of
+        lines. Then every line is colored with a different RGB(A) color.
+        If an array (N, ) is given, where N is the number of points then these
+        are considered as the values to be used by the colormap.
+        If an array (K,) is given, where K is the number of lines then these
+        are considered as the values to be used by the colormap.
+    lookup_colormap : vtkLookupTable, optional
+        Add a default lookup table to the colormap. Default is None which calls
+        :func:`fury.actor.colormap_lookup_table`.
+    linewidth : float, optional
+        Line thickness. Default is 1.
+
+    """
+    def __init__(self, directions, indices, values=None, affine=None,
+                 colors=None, lookup_colormap=None, linewidth=1):
+        if affine is not None:
+            w_pos = apply_affine(affine, np.asarray(indices).T)
+
+        valid_dirs = directions[indices]
+
+        num_dirs = len(np.nonzero(np.abs(valid_dirs).max(axis=-1) > 0)[0])
+
+        pnts_per_line = 2
+
+        points_array = np.empty((num_dirs * pnts_per_line, 3))
+        centers_array = np.empty_like(points_array, dtype=int)
+        diffs_array = np.empty_like(points_array)
+        line_count = 0
+        for idx, center in enumerate(zip(indices[0], indices[1], indices[2])):
+            if affine is None:
+                xyz = np.asarray(center)
+            else:
+                xyz = w_pos[idx, :]
+            valid_peaks = np.nonzero(
```

Comment:
@Garyfallidis I run some profiling tests here and these are the results:

For a `56x72x56` image the operation is adding less than 2 seconds (`0:00:01.707887`) to the for loop. Also, these are the average time and standard deviation of the operation: `Avg = 0:00:00.000017, Std = 0:00:00.000006`.

I don't think this is adding a significant delay in the peaks creation. But if you think something else needs to be done, please let me know.

Reply:
That image is quite small. You should try it with something much larger. Will merge this PR now. But to run the same test with  a much larger image. For example, the superresolved HYDI data.

Reply:
I ran the same test on an HCP 7T subject's peaks with a size of `176x207x173` and these are the results: The operation is adding less than 20 seconds (`0:00:19.486101`) to the for loop. The average time and standard deviation were similar to the ones obtained with the smaller image.

Additionally, I ran a comparison test between the for loop of the [current implementation](https://github.com/fury-gl/fury/blob/d498b6d1cf7b9594f55a80cfcbc047a9c8fd5bd1/fury/actor.py#L1175) of the peak slicer and the proposed version and found out that the peaks creation time of the new actor was around 20 minutes (`0:03:31.901080`) faster than the current version (`0:24:02.964906`).